### PR TITLE
Display searched details

### PR DESCRIPTION
**Pull request by rottencandy**
Displays fetched details from the redux store:

- Display message when no details are found:
empty
- Show spinner when details are being fetched:
loading
- Display fetched details using patternfly `Card`:
assign
- Show error message on failed request:
error

### CODE REVIEW

File: src/components/DetailSection.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import { RootState } from '../redux/reducers'
+import LoadingSpinner from './LoadingSpinner'
+import DisplayList from './DisplayList'
+import FailAlert from './FailAlert'
+
+type DisplaySectionProps = RootState['details'];
```
Comment (invincibleJai) [inline src/components/DetailSection.tsx]: this looks confusing, can we have a type defined here itself?

File: src/components/DetailSection.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+const DisplaySection: React.FC<DisplaySectionProps> = ({
+  isFetching,
+  isSuccessful,
+  data
+}) => {
+  if (isFetching) {
```
Comment (invincibleJai) [inline src/components/DetailSection.tsx]: can we use switch?

File: src/components/DetailSection.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    return <DisplayList data={data} />
+  }
+  return <FailAlert />
+}
+
+const mapStateToProps = (state: RootState): DisplaySectionProps => {
+  return state.details
+}
+
+export default connect(mapStateToProps)(DisplaySection)
```
Comment (invincibleJai) [inline src/components/DetailSection.tsx]: type for `mapStateToProps`?

File: src/components/MapDisplay.css [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#map {
```
Comment (invincibleJai) [inline src/components/MapDisplay.css]: any specific reason for  having styles with `ID`?

File: src/components/MapDisplay.css [hunk 5]
```diff
@@ -0,0 +1,4 @@
...
+    width: 500px;
+    height: 400px;
```
Comment (invincibleJai) [inline src/components/MapDisplay.css]: can make use of pf variables

File: src/components/MapDisplay.tsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import React, { useEffect } from 'react'
+import mapboxgl from 'mapbox-gl'
+import './MapDisplay.css'
+
+const LIGHT_RED = '#F86C6B'
```
Comment (invincibleJai) [inline src/components/MapDisplay.tsx]: make use of pf color

File: src/components/MapDisplay.tsx [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    const lngLat = [Number(lon), Number(lat)] as [number, number]
+    mapboxgl.accessToken = process.env.REACT_APP_MAPBOX_TOKEN || ''
+    const map = new mapboxgl.Map({
+      container: 'map',
+      style: 'mapbox://styles/mapbox/streets-v11',
+      center: lngLat,
+      zoom: 10
+    })
+    new mapboxgl.Marker({ color: LIGHT_RED }).setLngLat(lngLat).addTo(map)
+  })
```
Comment (invincibleJai) [inline src/components/MapDisplay.tsx]: when will this `useEffect` be triggered? can we have associated dependency?

File: src/components/DetailSection.tsx (hunk 1 above)
Comment (rottencandy) [reply src/components/DetailSection.tsx]: Added automatically inferenced types.

File: src/components/DetailSection.tsx (hunk 2 above)
Comment (rottencandy) [reply src/components/DetailSection.tsx]: Updated.

File: src/components/DetailSection.tsx (hunk 3 above)
Comment (rottencandy) [reply src/components/DetailSection.tsx]: Added.

File: src/components/MapDisplay.css (hunk 4 above)
Comment (rottencandy) [reply src/components/MapDisplay.css]: Mapbox API selects div element using it's ID.

File: src/components/MapDisplay.tsx (hunk 6 above)
Comment (rottencandy) [reply src/components/MapDisplay.tsx]: Updated.

File: src/components/MapDisplay.tsx (hunk 7 above)
Comment (rottencandy) [reply src/components/MapDisplay.tsx]: Added.

File: src/components/MapDisplay.css (hunk 5 above)
Comment (rottencandy) [reply src/components/MapDisplay.css]: Currently not feasible since we do not use SASS.

**Pull request merged**